### PR TITLE
add keyboardDismissBehavior into SingleChildScrollView

### LINKED ISSUE #76309: [proposal] provide a keyboardDismissBehavior on SingleChildScrollView
`ListView` has a `keyboardDismissBehavior` to customize how the keyboard should be closed when the list is being scrolled. Curently this feature is not available on `SingleChildScrollView` widget.

### Proposal

Add a `keyboardDismissBehavior` into `SingleChildScrollView`.

### PR DESCRIPTION
**Pull request by pedromassango**
Fixes https://github.com/flutter/flutter/issues/76309

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
This is a great contribution! Thanks @pedromassango!

File: packages/flutter/test/widgets/single_child_scroll_view_test.dart [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 import 'dart:ui';
 
+import 'package:flutter/material.dart';
+import 'package:flutter/src/material/text_field.dart';
```
[the remark this answers is not in the document]
Comment (pedromassango) [reply packages/flutter/test/widgets/single_child_scroll_view_test.dart]: Done. Thanks!

**Review by Piinks — APPROVED** [review]
LGTM! Thank you! 
I think you may need to rebase with master to resolve the plugins failure, it appears unrelated to this change.

**Comment by pedromassango** [comment]
✅

**Pull request merged**